docs/parameters.md

```diff
@@ -0,0 +1,322 @@
+  "parameters": {"ami_id":"ami-ecb68a84"}
+}' -H "Content-Type: application/json"
+```
+
+Note that the broker accepts an `ami_id` parameter ([AMI](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html) 
+identifier).
+To configure a provisioning request in Service Catalog, we need to declare an `Instance` 
+resource with an AMI identifier declared in the `parameters` field of its spec:
+```yaml
+apiVersion: servicecatalog.k8s.io/v1alpha1
+kind: Instance
+metadata:
+  name: ami-instance
+  namespace: test-ns
+spec:
+  serviceClassName: aws-ami
+  planName: default
+  parameters:
+    - name: ami_id
+      value: ami-ecb68a84
+```
+
+### Passing sensitive data
+
+To pass a parameter value stored in a `Secret` or a `ConfigMap`, use the `valueFrom` 
+field:
+```yaml
+apiVersion: servicecatalog.k8s.io/v1alpha1
+kind: Instance
+metadata:
+  name: qwerty-instance
+  namespace: test-ns
+spec:
+  serviceClassName: qwerty
+  planName: default
+  parameters:
+    - name: username
+      value: admin
+    - name: password
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: password
+```
+
+### Value types
+
+#### String
+
+By default any parameter value is treated as a string, and passed to the underlying 
+broker "as is", without further validation or processing. If the user needs to pass
+a value of different JSON data type, the field `type` is required to explicitly
+specify data type.
+
+#### JSON object
+
+`type: json` field value declares a value as a JSON object.
+```yaml
+apiVersion: servicecatalog.k8s.io/v1alpha1
+kind: Instance
+metadata:
+  name: qwerty-instance
+  namespace: test-ns
+spec:
+  serviceClassName: qwerty
+  planName: default
+  parameters:
+    - name: txt
+      value: Plain text, passed as a string
+    - name: meta
+      # Explicitly tells that the value is JSON which should be 
+      # validated and passed as a JSON object
+      type: json 
+      value: >
+          {
+            "attributes": [
+                {
+                  "type": "a",
+                  "A": "1",
+                },
+                {
+                  "type": "b",
+                  "B": "2",
+                },
+            ],
+            "labels": ["foo", "bar"]
+          }
+```
+
+If the JSON object is stored in a `Secret` or a `ConfigMap` key, the combination of 
+`valueFrom` and `type: json` fields allows to declare this as well:
+```yaml
+apiVersion: servicecatalog.k8s.io/v1alpha1
+kind: Instance
+metadata:
+  name: qwerty-instance
+  namespace: test-ns
+spec:
+  serviceClassName: qwerty
+  planName: default
+  parameters:
+    - name: txt
+      value: Plain text, passed as a string
+    - name: meta
+      # Explicitly tells that the value is JSON which should be 
+      # validated and passed as a JSON object
+      type: json
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: password
+```
+
+#### Other primitive types (int, bool, float)
```

Comment:
`float` might be an overkill, and it's unlikely to have float parameter at the top level, but if we want to support all kinds of JSON data types, we have to support it as well?